app/views/categories/search.html.erb

```diff
@@ -0,0 +1,79 @@
+<div class="container">
+
+<div class="row">
+  <h2><%= @category.name  %> Search Results</h2>
+
+  <% if @errors %>
+    <%= @errors %>
+  <% end %>
+</div>
+
+<div class="row">
+  <div class="col-md-6">
+    <h3>Existing Goals</h3>
+    <% index = 0 %>
+    <% if @database_results %>
+      <% @database_results.each do |result| %>
+        <div class="panel-group" role="tablist" aria-multiselectable="true">
+          <div class="panel panel-default">
+            <div class="panel-heading" role="tab" id="heading<%= index %>">
+              <h4 class="panel-title">
+                <a role="button" data-toggle="collapse" href="#collapse<%= index %>" aria-expanded="false" aria-controls="collapse<%= index %>">
+                  <%= result.task %>
+                </a>
+                <% if current_user.goals.include?(result) %>
+                  | <span>Goal is in your bucket list</span>
+                <% else %>
+                  | <span><%= link_to "Add to Goals", goals_users_path(goal_id: result.id), method: :post%></span>
+                <% end %>
```

Comment:
I'm surprised you have two different way of communicating whether or not the goal has been chosen by a user.

This also would make a great partial. One that decides which bucket icon to show.